src/isomorphism.rs

```diff
@@ -346,13 +346,49 @@ where
     new_graph
 }
 
-/// [Graph] Return `true` if the graphs `g0` and `g1` are isomorphic.
+trait SemanticMatcher<T> {
+    fn enabled(&self) -> bool;
+    fn eq(&mut self, _: &T, _: &T) -> PyResult<bool>;
+}
+
+struct NoSemanticMatch;
+
+impl<T> SemanticMatcher<T> for NoSemanticMatch {
+    #[inline]
+    fn enabled(&self) -> bool {
+        false
+    }
+    #[inline]
+    fn eq(&mut self, _: &T, _: &T) -> PyResult<bool> {
+        Ok(true)
+    }
+}
+
+impl<T, F> SemanticMatcher<T> for Option<F>
+where
+    F: FnMut(&T, &T) -> PyResult<bool>,
+{
+    #[inline]
+    fn enabled(&self) -> bool {
+        self.is_some()
+    }
+    #[inline]
+    fn eq(&mut self, a: &T, b: &T) -> PyResult<bool> {
+        let res = (self.as_mut().unwrap())(a, b)?;
+        Ok(res)
+    }
+}
```

Comment:
Is `NoSemanticMatch` actually being used anywhere? I know it pre-existed prior to this PR (and was originally just from the fork of petgraph's module) and you're just moving it around. But maybe we should just delete it if it's not needed.

Reply:
no, we can safely delete it.